### PR TITLE
Implementing  CatalogViewController

### PR DESCRIPTION
**Pull request by AnnaBanani**
Implementing  CatalogViewController

### CODE REVIEW

File: TheMET/TheMET/Services/FavoritesService.swift [hunk 1]
```diff
@@ -48,6 +48,11 @@ class FavoritesService {
     private (set) var favoriteArts: [Art] = []
     
     func addFavoriteArt(_ art: Art) {
+//        guard self.favoriteArts.contains(where: { favoriteArt in
```
Comment (Ssimboss) [inline TheMET/TheMET/Services/FavoritesService.swift:51]: Unused code - remove

File: TheMET/TheMET/ViewControllers/CategoryViewController.swift [hunk 2]
```diff
@@ -9,9 +9,17 @@ import Foundation
...
+    
+    private let metAPI = MetAPI(networkManager: NetworkManager.standard)
+    
     private let favoriteService = FavoritesService.standart
     
+    var departmentId: Int?
+    
+    private var imageLoader: ImageLoader = ImageLoader()
+    
+    private let favoritteService = FavoritesService.standart
```
Comment (Ssimboss) [inline TheMET/TheMET/ViewControllers/CategoryViewController.swift:21]: typo: `favoritteService` -> `favoritesService`

File: TheMET/TheMET/ViewControllers/CategoryViewController.swift [hunk 3]
```diff
@@ -83,44 +98,107 @@ class CategoryViewController: UIViewController, UITableViewDelegate, UITableView
...
     }
     
+    private func loadCellImage(cell: ArtViewCell, art: Art) {
+        cell.imageState = .loading
+        cell.tag = art.objectID
+        guard let imageURL =  art.primaryImage else {
+            cell.imageState = .failed
+            return
+        }
+        self.imageLoader.loadImage(urlString: imageURL) { image in
```
Comment (Ssimboss) [inline TheMET/TheMET/ViewControllers/CategoryViewController.swift:118]: make the `cell` weak in this completion and check it with `guard` as the first line

**Review by Ssimboss — CHANGES_REQUESTED** [review]

File: TheMET/TheMET/Services/FavoritesService.swift (hunk 1 above)
Comment (AnnaBanani) [reply TheMET/TheMET/Services/FavoritesService.swift:51]: +

File: TheMET/TheMET/ViewControllers/CategoryViewController.swift (hunk 2 above)
Comment (AnnaBanani) [reply TheMET/TheMET/ViewControllers/CategoryViewController.swift:21]: +

File: TheMET/TheMET/ViewControllers/CategoryViewController.swift (hunk 3 above)
Comment (AnnaBanani) [reply TheMET/TheMET/ViewControllers/CategoryViewController.swift:118]: +

**Review by Ssimboss — APPROVED** [review]

**Pull request merged**